### PR TITLE
Mcquadmj code review1

### PR DESCRIPTION
**Pull request by mcquadmj117**
Pull request for code review changes

### CODE REVIEW

File: src/main/java/com/quizapp/enterprise/persistence/GameTracker.java [hunk 1]
```diff
@@ -28,18 +28,21 @@ public void addGame(Game game){
...
+                .getPlayers()
+                .add(playerToJoin);
     }
 
     public static synchronized GameTracker getInstance() {
-
-        if (gameTracker == null) {
-            gameTracker = new GameTracker();
-        }
+        gameTracker = (gameTracker != null) ? gameTracker : new GameTracker();
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/persistence/GameTracker.java:45]: refactored the "if" statement that assigned a value based on a boolean expression to use the Java Ternary Operator.
This is recommend by the "Ternary Operator" section of the java coding best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/persistence/GameTracker.java [hunk 2]
```diff
@@ -28,18 +28,21 @@ public void addGame(Game game){
     }
 
     public Game getGameByCode(String gameCode){
-        return games.stream().filter(g -> g.getGameCode().equals(gameCode)).findAny().get();
+        return games.stream()
+                .filter(g -> g.getGameCode().equals(gameCode))
+                .findAny()
+                .get();
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/persistence/GameTracker.java:34]: reformatted the streams to use indentation so the code may be cleaner and easier to read.
This recommended by the "Iterators and Streams" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/config/WebSecurityAccessConfig.java [hunk 3]
```diff
@@ -48,16 +48,21 @@ protected void configure(AuthenticationManagerBuilder auth) throws Exception {
...
                 .and()
                 .formLogin()
                 .usernameParameter("email")
                 .defaultSuccessUrl("/quizzes")
                 .permitAll()
                 .and()
-                .logout().logoutSuccessUrl("/").permitAll();
+                .logout()
+                .logoutSuccessUrl("/")
+                .permitAll();
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/config/WebSecurityAccessConfig.java:65]: reformatted the streams to use indentation so the code may be cleaner and easier to read.
This recommended by the "Iterators and Streams" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/services/GameService.java [hunk 4]
```diff
@@ -16,7 +16,10 @@
     public Game startNewGame(int quizId) {
 
         Game newGame = new Game();
-        newGame.setGameCode(UUID.randomUUID().toString().substring(0,5));
+        newGame.setGameCode(UUID
+                .randomUUID()
+                .toString()
+                .substring(0, 5));
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/services/GameService.java:22]: reformatted the streams to use indentation so the code may be cleaner and easier to read.
This recommended by the "Iterators and Streams" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/services/GameService.java [hunk 5]
```diff
@@ -46,7 +49,13 @@ public void joinGame(String gameId, Player playerToJoin) throws Exception {
...
 
     private boolean userNameExists(String userName, String gameCode){
-        return GameTracker.getInstance().getGameByCode(gameCode).getPlayers().stream().anyMatch(player -> player.getPlayerUsername().equals(userName));
+        return GameTracker
+                .getInstance()
+                .getGameByCode(gameCode)
+                .getPlayers()
+                .stream()
+                .anyMatch(player -> player.getPlayerUsername()
+                        .equals(userName));
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/services/GameService.java:58]: reformatted the streams to use indentation so the code may be cleaner and easier to read.
This recommended by the "Iterators and Streams" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java [hunk 6]
```diff
@@ -64,19 +63,19 @@ void verifyAddAndRemoveGames() {
      */
     @Test
     void verifyCorrectAnswer() {
-        String userGuest = "Blubber";
+        String userGuess = "Blubber";
```
Comment (mcquadmj117) [inline src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java:66]: fixing typo

File: src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java [hunk 7]
```diff
@@ -64,19 +63,19 @@ void verifyAddAndRemoveGames() {
...
-                break;
+                fail("User Guess of "
+                        + guess.Letter
+                        + " was not correct");
+                return;
             }
         }
-
-        assertTrue(correct);
+        assertTrue(true);
```
Comment (mcquadmj117) [inline src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java:78]: Refactored test to use simplified "fail()" assertion when test fails and added message argument to allow the test to provide more valuable feedback in the exception should it fail.
This is recommended by the java ecosystem site referenced below in the "2.4 Testing Condition" section.

source: https://www.baeldung.com/junit-fail

File: src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java [hunk 8]
```diff
@@ -85,10 +84,10 @@ void verifyCorrectAnswer() {
      */
     @Test
     void verifyIncorrectAnswer() {
-        String userGuest = "Flubber";
+        String userGuess = "Flubber";
```
Comment (mcquadmj117) [inline src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java:87]: fixing typo

File: src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java [hunk 9]
```diff
@@ -85,10 +84,10 @@ void verifyCorrectAnswer() {
...
         String correctAnswer = "Blubber";
 
-        ArrayList<Guess> gec = new GameService().checkGuess(userGuest, correctAnswer);
+        ArrayList<Guess> gec = new GameService().checkGuess(userGuess, correctAnswer);
```
Comment (mcquadmj117) [inline src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java:90]: fixing typo

File: src/main/java/com/quizapp/enterprise/controllers/QuizController.java [hunk 10]
```diff
@@ -23,7 +23,7 @@
     @RequestMapping("/quizzes")
     public String userQuizes(Model model, Principal principal){
 
-        model.addAttribute("username",principal.getName());
+        model.addAttribute("username", principal.getName());
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/controllers/QuizController.java:26]: added space after argument within parameters for best practices and readability.
This is recommended by the "Spacing & line breaks" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/controllers/GameViewController.java [hunk 11]
```diff
@@ -18,7 +18,7 @@
     @RequestMapping("gameView/{gameId}")
     public String game(@PathVariable("gameId") String gameId, Model model){
 
-        model.addAttribute("players",gameService.getGame(gameId).getPlayers());
+        model.addAttribute("players", gameService.getGame(gameId).getPlayers());
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/controllers/GameViewController.java:21]: added space after argument within parameters for best practices and readability.
This is recommended by the "Spacing & line breaks" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/controllers/GameController.java [hunk 12]
```diff
@@ -56,12 +56,13 @@ public PlayerJoinEvent send(PlayerJoinEvent message) throws Exception {
         newPlayer.setPlayerUsername(message.getPlayerName());
         newPlayer.setHost(false);
         gameService.joinGame(message.getGameId(), newPlayer);
-        messagingTemplate.convertAndSend("/game1/messages/" + message.getGameId(),message);
+        messagingTemplate.convertAndSend("/game1/messages/" + message.getGameId(), message);
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/controllers/GameController.java:59]: added space after argument within parameters for best practices and readability.
This is recommended by the "Spacing & line breaks" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/controllers/GameController.java [hunk 13]
```diff
@@ -38,7 +38,7 @@ public Game getGame(@PathVariable("gameId") String gameId){
     }
 
     @PostMapping(value = "/joinGame/{gameId}")
-    public Game joinGame(@PathVariable("gameId") String gameId, @RequestBody Player player) throws Exception{
+    public Game joinGame(@PathVariable("gameId") String gameId, @RequestBody Player player) throws Exception {
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/controllers/GameController.java:41]: added space after method (before brackets) for best practices and readability.
This is recommended by the "Spacing & line breaks" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

File: src/main/java/com/quizapp/enterprise/persistence/GameTracker.java [hunk 14]
```diff
@@ -28,18 +28,21 @@ public void addGame(Game game){
...
     }
 
-    public void joinGame(String gameCode, Player playerToJoin){
-        getGameByCode(gameCode).getPlayers().add(playerToJoin);
+    public void joinGame(
+            String gameCode, Player playerToJoin)  {
+        getGameByCode(gameCode)
+                .getPlayers()
+                .add(playerToJoin);
```
Comment (mcquadmj117) [inline src/main/java/com/quizapp/enterprise/persistence/GameTracker.java:41]: reformatted the streams to use indentation so the code may be cleaner and easier to read.
This recommended by the "Iterators and Streams" section of the java best practices summary referenced below.

source: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3

**Comment by mcquadmj117** [comment]
**Code Review WriteUp/ Analysis**

**Description of Project**

- This project is an application that allows users to create a game by creating word-based puzzle/guessing prompts that would be presented to multiple participants (who join the game with a code) and then save this game to their account. The games saved in this application evaluate the participants answers and distribute points appropriately.

**Analysis Of The Program**

- The program was available on time in UC github
- The program was mostly understandable for an outsider like me to understand. Almost all classes, methods, and variables/properties were named in a way that made sense and gave me a good idea of what each piece of code was doing. There was some annotations and conventions that I did not recognize, but these were clearly just springboot/microservice concepts that we haven't covered just yet, but will be covering soon.
- At the time of the review, the program did not compile. However, this was only due to a "cannot resolve symbol" error for a service in the Unit Testing File, which you could argue is not a part of the core programs files. After commenting out or fixing this one unit test, the program compiled and ran fine.

**Three Concepts I Learned from this review**

- I learned how a java ternary operator can be used to not only conditionally assign values to a variable, but even help simplify code by replacing an "if" statement with a single line of code 
 -Ex:     gameTracker = (gameTracker != null) ? gameTracker : new GameTracker();

- I learned how streams that link properties, methods, or iterations together can be indented to improve readability within code
 -Ex: .filter(g -> g.getGameCode().equals(gameCode))
                .findAny()
                .get();
                .antMatchers("/quizzes")
                .authenticated()
                .antMatchers("/gameView/{gameId}")
                .permitAll()
                .anyRequest()
                .permitAll()
                .and()
                .formLogin()
                .usernameParameter("email")
                .defaultSuccessUrl("/quizzes")
                .permitAll()
                .and()
                .logout().logoutSuccessUrl("/").permitAll();
                .logout()
                .logoutSuccessUrl("/")
                .permitAll();

- I learned how the fail() assertion could be used to simplify a testing assertion and how a message can be passed in as an optional parameter to provide additional information within an exception
 -Ex:     if(!guess.IsCorrectLetter) {
                correct = false;
                break;
                fail("User Guess of "
                        + guess.Letter
                        + " was not correct");
                return;
            }


**Rational Behind My Changes**

- The rational behind my spacing and indentation changes in the code was based on the "Spacing & line breaks" and "Iterators and Streams" section of the java best practices summary here:  https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3. This sectioned explained that proper spacing and indentation is well worth implementing within code to provide better readability, and therefore time saved, for anybody reading or interacting with our code.
- The rational behind my ternary change was based on the "Ternary Operator" section of the java coding best practices summary: https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3. This section recommended ternaries based on their ability to simplify the code by reducing lines of code needed, and possibly eliminating "if" statements or other boolean conditional code.
- The rational behind my fail() assertion change was based on the "2.4 Testing Condition" section of the java ecosystem site: https://www.baeldung.com/junit-fail. The idea here is that we can simplify assertTrue() or assertFalse() conventions and provide even further feedback to the testing output/exceptions by passing in variables and messages to the assertion methods. These additional messages could give us a better idea of where/why the code is not working as expected.
- The rational behind my typo fixes was simply to provide better readability/understanding for the application/code

File: src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java (hunk 6 above)
Comment (discospiff) [reply src/test/java/com/quizapp/enterprise/EnterpriseApplicationTests.java:66]: Good.

**Comment by discospiff** [comment]
Some good things to consider here.  The group should review and decide what to harvest.

A note on formatting: The IDE can do this for you, with Control-Alt-Shift-L.  No need to create a new branch and do it manually.

**Pull request merged**